### ISSUE TITLE
[Feature Request]: Achievements Icon in Game List

### ISSUE THREAD

**Issue by mirrorsedger** [PCSX2/pcsx2#11404]
### Description

There are currently plenty of games that don't yet have achievements on RetroAchievements. I think it would be useful for users to add an icon, such as a trophy or similar symbol, to the game list in PCSX2 next to any game that has a set of achievements published on RetroAchievements. This will allow users to easily identify which games have available achievements. This feature could be enabled/disabled in global Achievement settings.

### Reason

This would provide an easier way to identify games with available achievements. For the user, this will save time otherwise spent manually checking each game's achievement status on the RetroAchievements website outside of PCSX2. Currently, there is no convenient method to see which games have achievements directly within the emulator without starting the game first, and this addition would fill that gap, offering a more seamless experience.

### Examples

Here is a poor example of what this could look like, but would obviously be better implemented:

![example](https://github.com/PCSX2/pcsx2/assets/19656591/9bbafac7-5886-4e2f-8a87-e182a317b7af)

**Comment by stenzek** [comment]
Sorry, not really possible with the way RA queries work. See https://github.com/PCSX2/pcsx2/issues/10208

tldr; it would require spamming their servers with requests to download achievement info, and then it'd end up stale if it was added later.
